src/algorithm/binarySearchTree.ts

```diff
@@ -0,0 +1,113 @@
+class TreeNode<Key, Val> {
+	public key: Key
+	public val: Val
+	public right: TreeNode<Key, Val>
+	public left: TreeNode<Key, Val>
+	constructor(key: Key, val: Val) {
+		this.key = key
+		this.val = val
+	}
+}
+
+export class binarySearchTree<Key, Val> {
```

Comment:
small things, but you can uppercase it 👍 
```ts
export class BinarySearchTree<Key, Val> {
```
the discussion is usually on function/methods, but classnames are uppercase even in JS codestyle 😄 

A really simplified version:
* class / enum / interface -> PascalCase
* everything else -> camelCase

More example : https://github.com/Platypi/style_typescript